mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1694,6 +1695,11 @@ void ModuleImport::processFunctionAttributes(llvm::Function *func,
         context, IntegerAttr::get(intTy, attr.getVScaleRangeMin()),
         IntegerAttr::get(intTy, attr.getVScaleRangeMax().value_or(0))));
   }
+  if (llvm::Attribute attr = func->getFnAttribute("target-features");
+      attr.isStringAttribute()) {
+    funcOp.setTargetFeaturesAttr(
+        LLVM::TargetFeaturesAttr::get(context, attr.getValueAsString()));
```

Comment:
nit: personally would find it clearer if `attr.isStringAttribute()` was in a separate if, threw me off a little.
```suggestion
  if (llvm::Attribute attr = func->getFnAttribute("target-features")) {
    if (attr.isStringAttribute())
      funcOp.setTargetFeaturesAttr(
          LLVM::TargetFeaturesAttr::get(context, attr.getValueAsString()));
  }
```

Reply:
I think it's LLVM style to avoid extra nesting, so I'm going keep this :)